app/src/main/java/com/droidhats/campuscompass/models/Map.kt

```diff
@@ -0,0 +1,72 @@
+package com.droidhats.campuscompass.models
+
+import androidx.fragment.app.FragmentActivity
+import com.droidhats.campuscompass.MainActivity
+import com.google.android.gms.maps.GoogleMap
+
+/**
+ * A Model for the map.
+ * This class has the objective of initializing the map and attaching listeners to it.
+ *
+ * @constructor Creates an initialized GoogleMap object.
+ * @param googleMap: A GoogleMap Object will be used to initialize the map.
+ * @param mapFragmentOnMarkerClickListener: a listener for marker clicks that will be attached to the map.
+ * @param mapFragmentOnPolygonClickListener: a listener for polygon clicks that will be attached to the map.
+ * @param activity: Used to check the location permission from the main activity.
+ */
+class Map(
+    var googleMap: GoogleMap,
+    var mapFragmentOnMarkerClickListener: GoogleMap.OnMarkerClickListener,
+    var mapFragmentOnPolygonClickListener: GoogleMap.OnPolygonClickListener,
+    var activity: FragmentActivity
+) {
+
+    companion object {
+        private const val MAP_PADDING_TOP = 200
+        private const val MAP_PADDING_RIGHT = 15
+    }
+
```

Comment:
why double space here?

Reply:
Thanks, I've removed the double space in my latest commit to this branch.